### PR TITLE
GraphQL: Add configured_variant attribute

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) adds the `configured_variant` to the `CartItemInterface`, per https://github.com/magento/magento2/pull/31380

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

### CODE REVIEW

**Review by dobooth — APPROVED** [review]

File: src/guides/v2.4/graphql/interfaces/cart-item-interface.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ The `ConfigurableCartItem` object adds the following attributes to the `CartItem
 Attribute | Data type | Description
 --- | --- | ---
 `configurable_options` | [[SelectedConfigurableOption!]!](#SelectedConfigurableOption) | An array of configurable options
+`configured_variant` | [ProductInterface]({{page.baseurl}}/graphql/interfaces/product-interface.html) | Product details of the cart item
```
Comment (cpartica) [inline src/guides/v2.4/graphql/interfaces/cart-item-interface.md:104]: there could be more to this:
 - regardless of the storeConfig value `configurable_thumbnail_source` which points to the child or parent while rendering cart the `configured_variant` will always show the child. It is up to the UI to query configured_variant when configurable_thumbnail_source is set to child. 
 There might other use cases when we want to know details about the `configured_variant` than just the thumbnail

**Review by cpartica — CHANGES_REQUESTED** [review]

File: src/guides/v2.4/graphql/interfaces/cart-item-interface.md (hunk 1 above)
Comment (cpartica) [reply src/guides/v2.4/graphql/interfaces/cart-item-interface.md:104]: also FYI this PR that you're linking to is obsolete, there's a new PR https://github.com/magento/magento2/pull/31380

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Review by cpartica — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**